src/parser/base.rs

```diff
@@ -212,11 +215,24 @@ impl<'a> Parser<'a> {
         }
     }
 
-    fn read_end(&mut self) {
+    fn read_end(&mut self) -> Option<()> {
         self.stream.advance();
+
+        let closing_tag_name = self.read_ident()?;
+
         self.read_to(b'>');
```

Comment:
Instead of using `read_ident()`, you could store the byte slice that is returned by `read_to(b'>')` in `closing_tag_name`, which will be whatever is between `</` and `>`. In valid HTML, this should always be the tag name (e.g. no spaces are allowed there).